assets/src/stories-editor/blocks/amp-story-page-attachment/attachment-content.js

```diff
@@ -129,15 +133,34 @@ const AttachmentContent = ( props ) => {
 					) }
 				</div>
 				<div className={ attachmentClass } style={ wrapperStyle }>
-					{ selectedPost && selectedPost.content && (
+					{ isFetching && <Spinner /> }
+					{ ! isFetching && selectedPost && selectedPost.content && (
 						<RawHTML>
 							{ `<h2>${ selectedPost.title.rendered }</h2>${ selectedPost.content.rendered }` }
 						</RawHTML>
 					) }
-					{ ( ! postId || failedToFetch ) && (
+					{ ! isFetching && ( ! postId || failedToFetch ) && (
 						<>
 							{ failedToFetch && (
-								<span>{ __( 'The selected post failed to load, please select a new post', 'amp' ) }</span>
+								<div>
+									{ ! isFetching && (
+										<>
+											<p className="failed-message">
+												{ __( 'The selected post failed to load, please select a new post or ', 'amp' ) }
+												<Button
+													className="refetch-attachment-post"
+													onClick={ ( event ) => {
+														event.stopPropagation();
+														fetchSelectedPost();
+													} }
+													isLink
+												>
+													{ __( 'try again', 'amp' ) }
```

Comment:
Unfortunately the tooling doesn't support string interpolation for translatable strings yet 😞 

However, string concatenation is not great for translators. All they would see is `The selected post failed to load, please select a new post or ` and a separate string `try again`, which makes it impossible to translate in every language.

Suggestion: use a full sentence `The selected post failed to load, please select a new post or  try again` and add a separate `Try Again` button after the sentence.

Reply:
Changes within 999e4fd.